perf/perf_stress_ng.py

```diff
@@ -0,0 +1,252 @@
+from avocado import Test
+from avocado.utils import process, build, archive
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+def clear_dmesg():
+    process.run("dmesg -C ", sudo=True)
+
+
+def collect_dmesg(object):
+    return process.system_output("dmesg").decode()
+
+
+class Stressng(Test):
+
+    """
+    Perf record with CPU stress
+    :source: git://kernel.ubuntu.com/cking/stress-ng.git
+    :param cpu_per: CPU Percentage to load
+    """
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.timeout = self.params.get('timeout', default='1')
+        self.cpu_per = self.params.get("cpu_load", default='10')
+        self.profile_dur = int(
+            self.params.get(
+                "profile_duration",
+                default='1'))
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            asset_url = 'https://github.com/ColinIanKing/stress-ng/archive/master.zip'
+            tarball = self.fetch_asset(
+                'stressng.zip', locations=[asset_url], expire='7d')
+            archive.extract(tarball, self.workdir)
+            sourcedir = os.path.join(self.workdir, 'stress-ng-master')
+            os.chdir(sourcedir)
+            result = build.run_make(
+                sourcedir, process_kwargs={
+                    'ignore_status': True})
+            for line in str(result).splitlines():
+                if 'error:' in line:
+                    self.cancel(
+                        "Build Failed, Please check the build logs for details !!")
+            build.make(sourcedir, extra_args='install')
+            clear_dmesg()
+        elif run_type == "distro":
+            deps = ['gcc', 'perf', 'stress-ng']
+            for package in deps:
+                if not smm.check_installed(
+                        package) and not smm.install(package):
+                    self.cancel(
+                        '%s is needed for the test to be run' %
+                        package)
+
+    def is_process_running(self, process_name):
+        for proc in psutil.process_iter(['pid', 'name']):
+            if process_name in proc.info['name']:
+                return True
+        return False
+
+    def kill_process_by_name(self, process_name):
+        for proc in psutil.process_iter(['pid', 'name']):
+            if process_name in proc.info['name']:
+                try:
+                    process = psutil.Process(proc.info['pid'])
+                    process.terminate()
+                    self.log.info(f"Process  %s terminated" % (process_name))
+                except psutil.NoSuchProcess:
+                    self.log.info(f"Process  %s not found" % (process_name))
+                    return
+
+    def vmstat_result(self, filename):
+        var = 0
+        sum = 0
+        result = 0
+        cmd = "awk '!/procs|swpd/ && NF > 0 { print $15 }' %s > /tmp/data1.txt" % filename
+        Perf_data = process.system_output(
```

Comment:
variable name in lower case having uniformity

Reply:
Done modified.